### DISCUSSION TITLE
Absence of track identities during single animal inferencing - not able to export "un-tracked" instances

### DISCUSSION THREAD

**Discussion by SabinNas** [talmolab/sleap#1659] [Help!]
Hello,

Some background: I'm using sleap to track a single animal in an arena with an occluded space (a covered area). Training has been working excellently, and inferencing is able to accurately find nodes, even when most of the body is occluded. I set occluded nodes as invisible and set all nodes as invisible when the entire body is occluded. I don't have many of these fully invisible skeletons as user-labelled instances because I'm not interested in tracking or having sleap label something in the space when the animal's not visible. I also assumed those fully invisible skeletons play little if any any role in training, but I wanted to mention it in case it's the cause of my error. 

As the title suggests, my issue is related to the absence of track identification in my predicted instances. All of my user labelled instances have a track value (1), but none of the inferred instances receive that value. Since I am only tracking one animal, I didn't expect to care about the track number value. However, frames with inferred instances (and lacking that track number) are omitted in the export process. My user-labelled instances (which have the track value) are present in the export files. I've exported to both csv and h5 and both export files just skip the frames that have no track number (ie the ones that are inferred).

I've been using the GUI and have tried playing with settings in the inference window, but nothing has resulted in any of the inferenced frames coming out with track numbers. I'm happy to provide more details if it will help with diagnostics. Thanks so much for your time.

**Comment by Marielamend** [comment]
Hello @SabinNas,

When exporting slp files, the instances are gathered into tracks but without tracks in the project, the export function assumes the project is only a single animal project. 

To fix this go to...
![image](https://github.com/talmolab/sleap/assets/120222245/28cc6478-a34d-4fc5-8fe8-7087fcef9a28)

Tracks > Delete Multiple Tracks... > All save, and retry the export!

Let us know if this helps,
Mariela

**Reply by roomrys** [reply]
Hi @SabinNas,

To add on here. If there are `Track`s in the project, then only `Instance`s assigned to `Track`s will be included in the exported file. Hence why you are only exporting your user-labeled `Instance`s which have the `Track` assigned.

By deleting all of the `Track`s in your project, SLEAP will assume that there is only a single animal per frame and export the first animal it finds in each `LabeledFrame` (skipping the task of organizing the `Instance`s by their assigned `Track` - which would be `None` for all your `PredictedInstance`s).

Thanks,
Liezl
